Extensions/CompositeMetadata.md

```diff
@@ -0,0 +1,45 @@
+# Composite Metadata Extension
+
+_This extension specification is currently incubating._
+
+## Introduction
+There are a number of situations where an arbitrary collection of discrete metadata types should be attached to frame.  For example, a request frame may want to include both routing metadata as well as tracing metadata.  This extension specification provides an interoperable structure for metadadata payloads to contain multiple discrete metadata types.  It is designed such that if a consumer of the metadata is unaware of a particular type, it can be safely skipped and the next one read.
+
+## Metadata Payload
+This metadata type **MUST** only be used in frame types used to initiate streams.  This includes [`REQUEST_FNF`][rf], [`REQUEST_RESPONSE`][rr], [`REQUEST_STREAM`][rs], and [`REQUEST_CHANNEL`][rc].  The [`SETUP` Frame][s] Metadata MIME Type is `message/x.rsocket.composite-metadata`.
+
+[rc]: ../Protocol.md#frame-request-channel
+[rf]: ../Protocol.md#frame-fnf
+[rr]: ../Protocol.md#frame-request-response
+[rs]: ../Protocol.md#frame-request-stream
+[s]:  ../Protocol.md#frame-setup
+
+### Metadata Contents
+```
+     0                   1                   2                   3
+     0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
+    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+    |            Version            |
+    +---------------+---------------+-------------------------------+
+    |  MIME Length  |   Metadata Encoding MIME Type                ...
+    +---------------+-------------------------------+---------------+
+    |              Metadata Length                  |
+    +-----------------------------------------------+---------------+
+    |                     Metadata Payload                         ...
+    +---------------+-----------------------------------------------+
+    |  MIME Length  |   Metadata Encoding MIME Type                ...
+    +---------------+-------------------------------+---------------+
+    |              Metadata Length                  |
+    +-----------------------------------------------+---------------+
+    |                     Metadata Payload                         ...
+    +---------------------------------------------------------------+
+                                   ...
+```
+
+* **Version**: (16 bits = max value 65,535) Unsigned 16-bit integer of version number of the composite metadata payload.
```

Comment:
2 bytes here seems like overkill.  Could either be part of the top level metadata mimetype when needed to be incremented, or possibly a single byte with space for flags?

Reply:
Yeah when I mentioned we should version this I didn’t about the mime-type for metadata - we could probably version this there and make that part of the spec.

Reply:
So is the consensus that we should have a MIME type of `message/x.rsocket.composite-metadata.v[\d]+` and remove the payload version?